pkg/volume/util/fs/fs_windows_test.go

```diff
@@ -17,65 +17,71 @@ limitations under the License.
 package fs
 
 import (
-	"fmt"
 	"io/ioutil"
 	"os"
+	"path/filepath"
 
-	"k8s.io/apimachinery/pkg/api/resource"
 	"testing"
 )
 
-func TestDiskUsage(t *testing.T) {
+func TestGetDirUsage(t *testing.T) {
 
 	dir_1, err := ioutil.TempDir("", "dir_1")
 	if err != nil {
-		t.Fatalf("TestDiskUsage failed: %s", err.Error())
+		t.Fatalf("Failed to create temp directory: %v", err)
 	}
 	defer os.RemoveAll(dir_1)
 
-	tmpfile_1, err := ioutil.TempFile(dir_1, "test")
-	if _, err = tmpfile_1.WriteString("just for testing"); err != nil {
-		t.Fatalf("TestDiskUsage failed: %s", err.Error())
+	file_1 := filepath.Join(dir_1, "test")
+	if err = ioutil.WriteFile(file_1, []byte("just for testing"), 0644); err != nil {
+		t.Fatalf("Failed to write test file %s: %v", file_1, err)
 	}
 	dir_2, err := ioutil.TempDir(dir_1, "dir_2")
 	if err != nil {
-		t.Fatalf("TestDiskUsage failed: %s", err.Error())
+		t.Fatalf("Failed to create temp directory: %v", err)
 	}
-	tmpfile_2, err := ioutil.TempFile(dir_2, "test")
-	if _, err = tmpfile_2.WriteString("just for testing"); err != nil {
-		t.Fatalf("TestDiskUsage failed: %s", err.Error())
+	file_2 := filepath.Join(dir_2, "test")
+	if err = ioutil.WriteFile(file_2, []byte("just for testing"), 0644); err != nil {
+		t.Fatalf("Failed to write test file %s: %v", file_2, err)
 	}
 
 	dirInfo1, err := os.Lstat(dir_1)
 	if err != nil {
-		t.Fatalf("TestDiskUsage failed: %s", err.Error())
+		t.Fatalf("Lstat failed for %s: %v", dir_1, err)
 	}
 	dirInfo2, err := os.Lstat(dir_2)
 	if err != nil {
-		t.Fatalf("TestDiskUsage failed: %s", err.Error())
+		t.Fatalf("Lstat failed for %s: %v", dir_2, err)
 	}
-	file1 := dir_1 + "/" + "test"
-	file2 := dir_2 + "/" + "test"
-	fileInfo1, err := os.Lstat(file1)
+
+	fileInfo1, err := os.Lstat(file_1)
 	if err != nil {
-		t.Fatalf("TestDiskUsage failed: %s", err.Error())
+		t.Fatalf("Lstat failed for %s: %v", file_1, err)
 	}
-	fileInfo2, err := os.Lstat(file2)
+	fileInfo2, err := os.Lstat(file_2)
 	if err != nil {
-		t.Fatalf("TestDiskUsage failed: %s", err.Error())
+		t.Fatalf("Lstat failed for %s: %v", file_2, err)
 	}
+
```

Comment:
based on the GetDisUsage implementation, isn't total should be equal to the returned usage size of dir1 because dir1 is the top level dir?

Reply:
Possibly, but I am not very familiar with how windows disk usage calculation works, and previous test did not test for equal:
```go
    if size.Cmp(used) != 1 {
        t.Fatalf("TestDiskUsage failed: %s", err.Error())
    }
```

I though there might be something I'm not aware of on windows, so I changed it to at least accept if they are equal so the test passes.

But Yeah, I think it should be equal (and it was on my test windows test vm)

Reply:
size.Cmp(used) != 1 means    size and used are not equal. So the original test is trying to test if size is not equal to used, it will fail.

Reply:
According the [documentation](https://godoc.org/k8s.io/apimachinery/pkg/api/resource#Quantity.Cmp) `size.Cmp(used) != 1` means that size is not greater than used.

> Cmp returns 0 if the quantity is equal to y, -1 if the quantity is less than y, or 1 if the quantity is greater than y.

Reply:
oh, my mistake. So here we should test equal size.Cmp(used) == 0?

Reply:
If the intention was to test the result to be equal, then `size.Cmp(used) == 0`. I can make the refactored test check for equal if you prefer that. I need to rebase it anyway.